HeroVault/HeroVault/CoreData/CoreDataHandler.swift

```diff
@@ -0,0 +1,196 @@
+//
+//  CoreDataHandler.swift
+//  HeroVault
+//
+//  Created by Gia Catano on 2024/04/30.
+//
+
+import CoreData
+import UIKit
+
+// MARK: - Entity Type Enum
+
+enum EntityType: String {
+    case character = "CoreDataCharacter"
+    case comic = "CoreDataComic"
+}
+
+// MARK: - Core Data Handler Protocol
+
+protocol CoreDataHandlerType {
+    func fetchAllObjectsFromCoreData(entityType: EntityType) -> [MarvelData]?
+    func saveObjectIntoCoreData(_ object: MarvelData)
+    func deleteObjectFromCoreData(_ object: MarvelData)
+    func deleteAllObjectsFromCoreData()
+    //delete this one
+    func showAllNames() -> [String]
+}
+
+// MARK: - Core Data Class
+
+class CoreDataHandler: CoreDataHandlerType {
+    
+    let appDelegate: AppDelegate
+    let context: NSManagedObjectContext
+    
+    init() {
+        guard let appDelegate = UIApplication.shared.delegate as? AppDelegate else {
+            fatalError("Unable to access AppDelegate")
+        }
+        
+        self.appDelegate = appDelegate
+        context = appDelegate.persistentContainer.viewContext
+    }
+    
+    // MARK: - Core Data CRUD functions
+    
+    func fetchAllObjectsFromCoreData(entityType: EntityType) -> [MarvelData]? {
+        if entityType == .character {
+            do {
+                return try context.fetch(CoreDataCharacter.fetchRequest())
+            } catch {
+                return nil
+            }
+        } else {
+            do {
+                return try context.fetch(CoreDataComic.fetchRequest())
+            } catch {
+                return nil
+            }
+        }
+    }
+    
+    func saveObjectIntoCoreData(_ object: MarvelData) {
+        let entityType: EntityType = object is Character ? .character : .comic
+        if !doesObjectExistInCoreData(object, entityType: entityType) {
+            createObjectInCoreData(object)
+            saveContext()
+            print("saved into core data: \(object.name)")
```

Comment:
I'm assuming you want to keep these print statements for future testing?

Reply:
Yes, I'll remove them once the favourites feature is completed and I no longer need to test